InternxtDesktop/Views/Backup/FolderSelectorView.swift

```diff
@@ -53,20 +47,36 @@ struct FolderSelectorView: View {
                         panel.canChooseFiles = false
                         if panel.runModal() == .OK {
                             for url in panel.urls {
-                                let foldername = url.lastPathComponent
-                                self.urls.append(url.absoluteString)
-                                self.foldernames.append(foldername)
+                                do {
+                                    if (!backupsService.urls.contains(url)) {
+                                        try self.backupsService.addFoldernameToBackup(
+                                          FoldernameToBackup(
+                                              url: url.absoluteString,
+                                              status: .selected
+                                          )
+                                        )
+                                    } else {
+                                        showErrorDialog(message: "Already added folder")
+                                    }
+                                } catch {
+                                    // show error in UI
+                                    showErrorDialog(message: error.localizedDescription)
```

Comment:
Be careful with this, as we don't want to catch all the errors and display them in the UI, users should see an human friendly error, and while `localizedDescription` usually provides human readable messages, is not always the case, no changes needed here, but just be careful and don't overuse this pattern